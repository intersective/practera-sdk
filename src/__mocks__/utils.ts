export const isEmpty = jest.fn();

export const has = jest.fn();

export const urlFormatter = jest.fn();
