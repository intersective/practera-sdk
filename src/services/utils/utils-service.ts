/**
 * Method will combine the API url with endpoint and create full URL for request.
 * @param apiUrl string - Api actual URL
 * @param endpoint string - Api endpoint
 * @returns string - Full API URL.
 */
export const createFullApiUrl = (apiUrl: string, endPoint: string): string => {
  if (!apiUrl || !endPoint) {
    console.error('apiUrl and endPoint is required');
  }
  return urlFormatter(apiUrl, endPoint);
};

/**
 * Given a domain and endpoint string, return an absolute url formatted string
 *
 * e.g.
 * test.practera.com - login => https://test.practera.com/login
 *
 * @name urlFormatter
 * @param {string} domain url
 * @param {string} endpoint optional relative url
 * @return {string}   combination of scheme+domain+endpoint
 */
export const urlFormatter = (domain: string, endpoint?: string): string => {
  // always need http as prefix
  let theDomain = !domain.match(/^http/) ? `https://${domain}` : domain;
  // remove / in suffix
  theDomain = theDomain.replace(/\/$/, '');
  if (!endpoint) {
    return theDomain;
  }
  // always have / in prefix
  let theEndpoint = !endpoint.match(/^\//) ? `/${endpoint}` : endpoint;
  // remove / in suffix
  theEndpoint = theEndpoint.replace(/\/$/, '');
  return `${theDomain}${theEndpoint}`;
};
