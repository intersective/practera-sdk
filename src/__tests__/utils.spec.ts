import { createFullApiUrl, isEmpty, has } from '../utils';
import _ from "lodash";

describe('When testing createFullApiUrl()', (): void => {
  it('should return full API URL', (): void => {
    const domain = 'testAPI.com';
    const fullUrl: string = createFullApiUrl(`https://${domain}/`, 'login');
    expect(fullUrl).toEqual(`https://${domain}/login`);
  });
  it('should console error if apiUrl empty', (): void => {
    spyOn(console, 'error');
    const fullUrl: string = createFullApiUrl('', 'login');
    expect(console.error).toBeCalled();
  });
  it('should console error if endPoint empty', (): void => {
    spyOn(console, 'error');
    const fullUrl: string = createFullApiUrl('testAPI.com/', '');
    expect(console.error).toBeCalled();
  });
});

describe('isEmpty()', () => {
  it('should check if target parameter is empty (undefined, null, {}, \'\')', () => {
    expect(isEmpty('')).toBeTruthy();
    expect(isEmpty({})).toBeTruthy();
    expect(isEmpty(null)).toBeTruthy();
    expect(isEmpty(undefined)).toBeTruthy();
    expect(isEmpty(0)).toBeFalsy();
    expect(isEmpty(1)).toBeFalsy();
  });
});

describe('lodash extensions', () => {
  it('should has', () => {
    spyOn(_, 'has');
    has([1, 2, 3], () => true);
    expect(_.has).toHaveBeenCalled();
  });
});
