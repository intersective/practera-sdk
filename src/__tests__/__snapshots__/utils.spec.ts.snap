// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When testing urlFormatter() should return full API URL: "http://test.practera.com", "/login" 1`] = `"http://test.practera.com/login"`;

exports[`When testing urlFormatter() should return full API URL: "http://test.practera.com", "login" 1`] = `"http://test.practera.com/login"`;

exports[`When testing urlFormatter() should return full API URL: "http://test.practera.com", "login/" 1`] = `"http://test.practera.com/login"`;

exports[`When testing urlFormatter() should return full API URL: "http://test.practera.com", "undefined" 1`] = `"http://test.practera.com"`;

exports[`When testing urlFormatter() should return full API URL: "http://test.practera.com/", "/login" 1`] = `"http://test.practera.com/login"`;

exports[`When testing urlFormatter() should return full API URL: "http://test.practera.com/", "login" 1`] = `"http://test.practera.com/login"`;

exports[`When testing urlFormatter() should return full API URL: "http://test.practera.com/", "login/" 1`] = `"http://test.practera.com/login"`;

exports[`When testing urlFormatter() should return full API URL: "http://test.practera.com/", "undefined" 1`] = `"http://test.practera.com"`;

exports[`When testing urlFormatter() should return full API URL: "https://test.practera.com", "/login" 1`] = `"https://test.practera.com/login"`;

exports[`When testing urlFormatter() should return full API URL: "https://test.practera.com", "login" 1`] = `"https://test.practera.com/login"`;

exports[`When testing urlFormatter() should return full API URL: "https://test.practera.com", "login/" 1`] = `"https://test.practera.com/login"`;

exports[`When testing urlFormatter() should return full API URL: "https://test.practera.com", "undefined" 1`] = `"https://test.practera.com"`;

exports[`When testing urlFormatter() should return full API URL: "https://test.practera.com/", "/login" 1`] = `"https://test.practera.com/login"`;

exports[`When testing urlFormatter() should return full API URL: "https://test.practera.com/", "login" 1`] = `"https://test.practera.com/login"`;

exports[`When testing urlFormatter() should return full API URL: "https://test.practera.com/", "login/" 1`] = `"https://test.practera.com/login"`;

exports[`When testing urlFormatter() should return full API URL: "https://test.practera.com/", "undefined" 1`] = `"https://test.practera.com"`;

exports[`When testing urlFormatter() should return full API URL: "test.practera.com", "/login" 1`] = `"https://test.practera.com/login"`;

exports[`When testing urlFormatter() should return full API URL: "test.practera.com", "login" 1`] = `"https://test.practera.com/login"`;

exports[`When testing urlFormatter() should return full API URL: "test.practera.com", "login/" 1`] = `"https://test.practera.com/login"`;

exports[`When testing urlFormatter() should return full API URL: "test.practera.com", "undefined" 1`] = `"https://test.practera.com"`;

exports[`When testing urlFormatter() should return full API URL: "test.practera.com/", "/login" 1`] = `"https://test.practera.com/login"`;

exports[`When testing urlFormatter() should return full API URL: "test.practera.com/", "login" 1`] = `"https://test.practera.com/login"`;

exports[`When testing urlFormatter() should return full API URL: "test.practera.com/", "login/" 1`] = `"https://test.practera.com/login"`;

exports[`When testing urlFormatter() should return full API URL: "test.practera.com/", "undefined" 1`] = `"https://test.practera.com"`;
